components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java

```diff
@@ -516,6 +541,29 @@ public List<Application> getAllApplications(String organization) {
         return applications;
     }
 
+    /*
+     * This method can be used to retrieve all the Applications of a given tenant in the database
+     * @param tenantId : tenant Id
+     * @param systemOrg: organization Id of system org
+     * @return {@link Subscription}
+     * */
+    public List<Application> getAllApplications(String organization, String systemOrg) {
+
+        ArrayList<Application> applications = new ArrayList<>();
+        try (Connection conn = APIMgtDBUtil.getConnection();
+             PreparedStatement ps =
+                     conn.prepareStatement(SubscriptionValidationSQLConstants.GET_APPLICATIONS_BY_ORGANIZATION_SQL_WITH_SYSTEM_APPS)) {
+            ps.setString(1, organization);
+            ps.setString(2,systemOrg);
+            try (ResultSet resultSet = ps.executeQuery()) {
+                addToApplicationList(applications, resultSet);
+            }
+        } catch (SQLException e) {
+            log.error("Error in loading Applications for organization : " + organization + " including system org", e);
```

Comment:
Let's throw the errors. Lets fix the other places too

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12101/commits/202fb123e6ff5128e5b52ec84b71d792457a2119